File: Ejercicio1.hs
-- Verificador de edad legal - Haskell

-- Función pura que devuelve un String
mayorDeEdad :: Int -> String
mayorDeEdad edad
  | edad < 0  = "Edad inválida: no puede ser negativa."
  | edad >= 18 = "Eres mayor de edad."
  | otherwise  = "Eres menor de edad."

-- Ejemplo de main para uso por consola (opcional)
main :: IO ()
main = do
  putStrLn "Ingresa tu edad:"
  linea <- getLine
  case reads linea :: [(Int, String)] of
    [(edad, "")] -> putStrLn (mayorDeEdad edad)
    _            -> putStrLn "Entrada inválida: por favor ingresa un número entero."

File: Ejercicio1.py
# Verificador de edad legal - Python
def mayor_de_edad(edad):
    return edad >= 18

def main():
    entrada = input("Ingresa tu edad: ").strip()
    try:
        edad = int(entrada)
        if edad < 0:
            print("Edad inválida: no puede ser negativa.")
            return
    except ValueError:
        print("Entrada inválida: por favor ingresa un número entero.")
        return

    if mayor_de_edad(edad):
        print("Eres mayor de edad.")
    else:
        print("Eres menor de edad.")

if __name__ == "__main__":
    main()
